quantestpy/assertion/assert_equivalent_counts.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+from typing import Dict, Union
+
+import numpy as np
+
+from quantestpy.exceptions import QuantestPyAssertionError
+
+ut_test_case = unittest.TestCase()
+
+
+def assert_equivalent_counts(
+        counts_a: Dict[str, int],
+        counts_b: Dict[str, int],
+        sigma: Union[float, int] = 2.,
+        msg=None) -> None:
+
+    if not isinstance(counts_a, dict) or not isinstance(counts_b, dict):
+        raise TypeError(
+            "The type of counts must be dict."
+        )
+
+    if not isinstance(sigma, float) and not isinstance(sigma, int):
+        raise TypeError(
+            "The type of sigma must be float or int."
+        )
+
+    for k, v in counts_a.items():
+        if not isinstance(k, str):
+            raise TypeError(
+                "The type of key in counts must be str."
+            )
+        if not isinstance(v, int):
+            raise TypeError(
+                "The type of value in counts must be int."
+            )
+        if v < 0:
+            raise ValueError(
+                "The value in counts must be non-negative."
+            )
+
+    for k, v in counts_b.items():
+        if not isinstance(k, str):
+            raise TypeError(
+                "The type of key in counts must be str."
+            )
+        if not isinstance(v, int):
+            raise TypeError(
+                "The type of value in counts must be int."
+            )
+        if v < 0:
+            raise ValueError(
+                "The value in counts must be non-negative."
+            )
+
+    if len(counts_a) != len(counts_b):
+        err_msg = "The number of keys in counts are not the same."
+        msg = ut_test_case._formatMessage(msg, err_msg)
+        raise QuantestPyAssertionError(msg)
+
+    for k in counts_a.keys():
+        if k in counts_b.keys():
+            v_a = counts_a[k]
+            v_b = counts_b[k]
+            err_a = np.sqrt(v_a)
+            err_b = np.sqrt(v_b)
+
+            diff = np.abs(v_a - v_b)
+            tole = (err_a + err_b) * sigma
```

Comment:
Why did you define it this way?
I think this definition is unsuitable because the dimension of 'diff' is different from that of 'tole'.
Of course, `count` has no dimension in this case, but please image the case when the dimension of `v_a` is length(`cm`).

If there are a lot of data, we may use `Root Sum Squire` as the definition of 'tole'. 
But I don't know an appropriate definition when we compare two values...

Reply:
It is assumed that `counts_a[key]` and `counts_b[key]` are always the number of times each bitstring `key` was measured. I believe that users can understand it from the document `doc/assertion/assert_equivalent_counts.md`. 

The standard deviation for the number of times measured `N` is `sqrt(N)`, therefore the definition of `tole` should be okay. Another possibility may be
```
tole = sqrt(v_a + v_b) * sigma
```
It is fine for me to define the both and leave users to decide which one they use?